print-ospf.c

```diff
@@ -803,15 +803,16 @@ ospf_print_ep_lsa_extd_prefix_tlv(netdissect_options *ndo, const uint8_t *tptr,
 	    algo = GET_U_1(tptr+3);
 
 	    if (subtlv_length == 7) {
-		ND_PRINT("\n\t\t  Label: %u, MT-ID: %u, Algorithm: %u",
-			 GET_BE_U_3(tptr+4), mt_id, algo);
+		ND_PRINT("\n\t\t  Label: %u", GET_BE_U_3(tptr+4));
 	    } else if (subtlv_length == 8) {
-		ND_PRINT("\n\t\t  Index: %u, MT-ID: %u, Algorithm: %u, Flags [%s]",
-			 GET_BE_U_4(tptr+4), mt_id, algo,
-			 bittok2str(ep_range_tlv_prefix_sid_subtlv_flag_values, "none", flags));
+		ND_PRINT("\n\t\t  Index: %u", GET_BE_U_4(tptr+4));
 	    } else {
 		ND_PRINT("\n\t\tBogus subTLV length %u", subtlv_length);
+		break;
 	    }
+	    ND_PRINT( ", MT-ID: %u, Algorithm: %s (%u), Flags [%s]",
+		      mt_id, tok2str(lsa_opaque_ri_tlv_sr_algos, "Unknown", algo), algo,
+		 bittok2str(ep_range_tlv_prefix_sid_subtlv_flag_values, "none", flags));
```

Comment:
@hannesgredler just wanted to check that you're ok with this change to your prefix printing - using the list of sr algorithms to print `Shortest-Path First (0)` instead of `0`.

Reply:
Yes, that's fine.